### PR TITLE
Generate package READMEs

### PR DESCRIPTION
**Pull request by karlkfi**
- Add kpt-pkg-readme.sh to generate a readme
- Add fix-readmes.sh to re-generate all readmes
- Add and update Kptfile descriptions

Sections:
- **Title** - from `.metadata.annotations["blueprints.cloud.google.com/title"]`
- **Description** - from `.info.description` (supports multi-line comments with markdown)
- **Setters** - from kpt-list-setters.sh (includes both setters.yaml fields and kpt-set comments)
- **Sub-packages** - lists nearest sub-packages that contain a Kptfile
- **Resources** - table of resources with File, APIVersion, Kind, Name, and Namespace
- **Resource References** - links to resource reference docs (Supports KCC resources, K8s resources, ConfigConnectorContext, and ConfigManagement)
- **Usage** - basic instructions to apply with kpt

Function ConfigPaths and invisible files/dirs are ignored from all sections except Setters.

```
$ ./utils/fix-readmes.sh 
Generating catalog/networking/firewall/common-rules/README.md
Generating catalog/networking/svpc-service-project/README.md
Generating catalog/networking/vpc-service-controls/perimeter/README.md
Generating catalog/networking/vpc-service-controls/access-policy/README.md
Generating catalog/networking/peering/README.md
Generating catalog/networking/network/README.md
Generating catalog/networking/network/vpc/README.md
Generating catalog/networking/network/subnet/README.md
Generating catalog/networking/shared-vpc/README.md
Generating catalog/networking/dns/managedzone-private/README.md
Generating catalog/networking/dns/managedzone-peering/README.md
Generating catalog/networking/dns/recordset/README.md
Generating catalog/networking/dns/policy/README.md
Generating catalog/networking/dns/managedzone-forwarding/README.md
Generating catalog/networking/routes/routes-igw/README.md
Generating catalog/bucket/README.md
Generating catalog/spanner/README.md
Generating catalog/gke/README.md
Generating catalog/gke/cluster/README.md
Generating catalog/gke/subnet/README.md
Generating catalog/gke/nodepools/primary/README.md
Generating catalog/redis-bucket/README.md
Generating catalog/gitops/README.md
Generating catalog/gitops/configsync/README.md
Generating catalog/log-export/org/storage-export/README.md
Generating catalog/log-export/org/pubsub-export/README.md
Generating catalog/log-export/org/bigquery-export/README.md
Generating catalog/log-export/folder/storage-export/README.md
Generating catalog/log-export/folder/pubsub-export/README.md
Generating catalog/log-export/folder/bigquery-export/README.md
Generating catalog/hierarchy/team/README.md
Generating catalog/hierarchy/env-bu/README.md
Generating catalog/hierarchy/simple/README.md
Generating catalog/hierarchy/bu/README.md
Generating catalog/landing-zone/README.md
Generating catalog/project/README.md
```

### CODE REVIEW

**Comment by karlkfi** [comment]
Blocking on https://github.com/GoogleCloudPlatform/blueprints/pull/48 to fixup gitops & configsync pkg.

**Comment by karlkfi** [comment]
Ready for review.

If you don't like a name or description, please recommend a new one. 

As I understand it, the name in the Kptfile is largely irrelevant to the user because `kpt pkg get` uses the directory name by default. But if that's incorrect, let me know. In the future hopefully we can use a title field instead (go/blueprint-metadata).

The readme generator is a lot more code than I expected at first, because I kept adding new sections and features (like filtering configPaths and supporting inline configMaps). Hopefully my bash is defensive enough to be debuggable and maintainable, but we can always rewrite it as a kpt function later, if that's preferred. Using python or go would probably make the yaml parsing easier and avoid the yq/jq dependency.

Don't let perfect be the enemy of good.

**Review by morgante — CHANGES_REQUESTED** [review]
Please do not make the titles in READMEs *less* descriptive than they already are.

I don't want to make the perfect the enemy of the good here. We are fine to include this as an autogeneration script for initial READMEs. I think that's certainly helpful.

We will work on refining it in the future to handle human-enhanced names, but for the time being please leave existing titles untouched. I'd rather not have to go back and add them back later.

File: catalog/redis-bucket/README.md [hunk 1]
```diff
@@ -1,4 +1,74 @@
-# Google Cloud Redis and Storage bucket blueprint
+# Redis Bucket blueprint
```
Comment (karlkfi) [inline catalog/redis-bucket/README.md:1]: Not sure where "Google Cloud Redis" came from. I think the product name is "Memorystore for Redis". "Redis Bucket" isn't really an improvement. 

What should this actually be named? "Memorystore for Redis with Cloud Storage" is kinda long... maybe just "Memorystore for Redis" ?

Comment (morgante) [reply catalog/redis-bucket/README.md:1]: How about `Google Cloud Storage and Memorystore for Redis`? It's wordy but the most descriptive. I think I would be surprised for a GCS bucket to show up in a blueprint just called `Memorystore for Redis`.

Comment (karlkfi) [reply catalog/redis-bucket/README.md:1]: sold.

**Comment by karlkfi** [comment]
Replaced by https://github.com/GoogleCloudPlatform/blueprints/pull/52

**Pull request closed without merging**